app/scripts/modules/core/src/managed/ToggleManagedResourceForApplication.tsx

```diff
@@ -0,0 +1,130 @@
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  showModal,
+} from '../presentation';
+import { ManagedWriter } from './ManagedWriter';
+
+const logClick = (label: string, application: string) =>
+  ReactGA.event({
+    category: 'Managed Resource Environments',
+    action: `${label} clicked`,
+    label: application,
+  });
+
+export interface IToggleManagedResourceForApplicationModalProps extends IModalComponentProps {
+  application: Application;
+}
+
+interface Error {
+  data: { error: string; message: string };
+}
+
+export const MDRunningStateDescription = () => (
+  <>
+    <p>
+      <span className="bold">
+        Careful! You’re about to stop Spinnaker from managing all resources in your application.
+      </span>
+      This feature should only be used if management is not working properly and manual intervention is required.{' '}
+      <a href="https://www.spinnaker.io/guides/user/managed-delivery" target="_blank">
+        Check our documentation for more information
+      </a>
+      .
+    </p>
+    <p>
+      Do you just need to rollback?{' '}
+      <a href="https://www.spinnaker.io/guides/user/managed-delivery/pinning/" target="_blank">
+        Please use the “pin” feature instead
+      </a>
+      .
+    </p>
+  </>
+);
+
+export const MDPausedStateDescription = () => {
+  return (
+    <p>
+      You’re about to resume management of cloud resources for your application. The latest good version approved for
+      deployment will be deployed to each environment.
+    </p>
+  );
+};
+
+export const showToggleManagedResourceModal = (props: IToggleManagedResourceForApplicationModalProps) =>
+  showModal(ToggleManagedResourceForApplicationModal, props, { maxWidth: 800 });
+
+export const ToggleManagedResourceForApplicationModal = memo(
+  ({ application, closeModal, dismissModal }: IToggleManagedResourceForApplicationModalProps) => {
+    const [isSubmitting, setSubmitting] = useState<boolean>(false);
+    const [actionError, setActionError] = useState<{ error: string; message: string } | null>(null);
+    const dataSource = application.getDataSource('environments');
+    let actionToBeTaken = 'Disable';
+    let description = <MDRunningStateDescription />;
+    let icon: IllustrationName = 'disableManagement';
+
+    if (application.isManagementPaused) {
+      actionToBeTaken = 'Resume';
+      description = <MDPausedStateDescription />;
+      icon = 'runManagement';
+    }
+
+    const handleActionInitiation = () => {
+      setActionError(null);
+      setSubmitting(true);
+      logClick(`${actionToBeTaken} Management`, application.name);
+      const call = application.isManagementPaused
+        ? ManagedWriter.resumeApplicationManagement(application.name)
+        : ManagedWriter.pauseApplicationManagement(application.name);
+      call
+        .then(() => dataSource.refresh(true).catch(() => null))
+        .then(() => {
+          application.isManagementPaused = actionToBeTaken === 'Disable';
```

Comment:
in other places, we're not doing any in-memory updates and relying entirely on refreshing the data source to get fresh app state. Are you seeing a lag here? (i.e. you don't see an updated `applicationPaused` value in the data from keel on the first refresh?)

Reply:
The lag was significant and it wasn't due to `keel` not responding in time. As I was quickly debugging to explain the issue, I actually found the cause of the problem 🤦 .

`application.isManagementPaused` is only [set](https://github.com/spinnaker/deck/blob/05ec5eff04ad4d114ed21dbaafad1adbd4b23e42/app/scripts/modules/core/src/managed/managed.dataSource.ts#L27) when we refresh `managedResources` data source.

If I updated the data source in this component to `managedResources`, it would set `application.isManagementPaused` right after keel updates it. But the data source in the top level `Environments` component is for [`environments`](https://github.com/spinnaker/deck/blob/05ec5eff04ad4d114ed21dbaafad1adbd4b23e42/app/scripts/modules/core/src/managed/Environments.tsx#L69). So the top-level component is not updated for a while, so we continue to see stale information.

I've fixed it by adding `application.isManagementPaused` in the [environments](https://github.com/spinnaker/deck/blob/05ec5eff04ad4d114ed21dbaafad1adbd4b23e42/app/scripts/modules/core/src/managed/managed.dataSource.ts#L45) data source. Let me know if that is the right fix.

Reply:
wow nice find! geez. I remember back when Alan and I discussed having these two data sources we were pretty sure there would be some obnoxious state-syncing silliness going on as we built more UI surfaces out, apparently it's starting to happen. I'm hopeful as we start experimentally moving this area of the UI over to GraphQL we'll get away from problems like this, but if it continues to prove problematic we might need to do some sort of data source joinery between the two. I think the way you went about it is totally reasonable for now, but I imagine in the future it's possible we'll end up with racing or some other random annoyances.